File: Dagger2TestAndroid/app/src/main/java/com/example/raghu/dagger2testandroid/ui/main/MainActivityModule.java
package com.example.raghu.dagger2testandroid.ui.main;

import com.example.raghu.dagger2testandroid.data.MainModel;
import com.example.raghu.dagger2testandroid.presenter.MainActivityPresenter;
import com.example.raghu.dagger2testandroid.presenter.MainPresenterContract;

import dagger.Module;
import dagger.Provides;

/**
 * Created by mertsimsek on 25/05/2017.
 */
@Module
public class MainActivityModule {

    @Provides
    MainPresenterContract.View provideMainView(MainActivity mainActivity){
        return mainActivity;
    }

    @Provides
    MainActivityPresenter provideMainPresenter(MainPresenterContract.View  mainView, MainModel mainModel){
        return new MainActivityPresenter(mainView, mainModel);
    }
}

File: SpecBeeMapAssignment/app/src/main/java/com/example/raghu/specbeemapassignment/dagger/MainActivityModule.kt
package com.example.raghu.specbeemapassignment.dagger

import com.example.raghu.dagger2testandroid.data.MainModel
import com.example.raghu.dagger2testandroid.presenter.MainActivityPresenter
import com.example.raghu.dagger2testandroid.presenter.MainPresenterContract
import com.example.raghu.specbeemapassignment.MainActivity

import dagger.Module
import dagger.Provides

/**
 * Created by mertsimsek on 25/05/2017.
 */
@Module
class MainActivityModule {

    @Provides
    internal fun provideMainView(mainActivity: MainActivity): MainPresenterContract.View {
        return mainActivity
    }

    @Provides
    internal fun provideMainPresenter(mainView: MainPresenterContract.View, mainModel: MainModel): MainActivityPresenter {
        return MainActivityPresenter(mainView, mainModel)
    }
}
